File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/model/dto/request/process/CoreProcessRequestDTO.java
package org.infra.reactive.form.engine.form.engine.model.dto.request.process;

import lombok.Data;
import org.infra.reactive.form.engine.form.engine.model.dto.response.element.CoreAllElementDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.translate.CoreTranslateLanguageDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.window.tab.RecordModelDTO;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

@Data
public class CoreProcessRequestDTO implements Serializable {
    private String uuid;
    private Long id;
    private CoreTranslateLanguageDTO coreTranslateLanguageDTO;
    private Map<Long, Serializable> coreProcessParamValueMap;
    private List<RecordModelDTO> recordModelDTOList;
    private Long recordId;
    private CoreAllElementDTO coreAllElementDTO;
    private CoreProcessRequestStatusDTO coreProcessRequestStatusDTO;
}

File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Request/Process/CoreProcessRequestDTO.js
import {BaseModel} from "../../../../../UIFrameWork/Shared/Common/BaseModel.js";
import ConvertUtil from "../../../Common/ConvertUtil.js";
import CoreTranslateLanguageDTO from "../../Response/Translate/CoreTranslateLanguageDTO.js";
import CoreProcessRequestStatusDTO from "./CoreProcessRequestStatusDTO.js";
import RecordModelDTO from "../../Response/Window/Tab/RecordModelDTO.js";

export default class CoreProcessRequestDTO extends BaseModel {
    constructor() {
        super();
    }

    setCoreProcessParamValueMap(coreProcessParamValueMap) {
        this.coreProcessParamValueMap = coreProcessParamValueMap;
    }

    getCoreTranslateLanguageDTO() {
        if (this.coreTranslateLanguageDTO && !(this.coreTranslateLanguageDTO.constructor instanceof BaseModel)) {
            this.coreTranslateLanguageDTO = ConvertUtil.ConvertGeneral(CoreTranslateLanguageDTO, this.coreTranslateLanguageDTO);
        }
        return this.coreTranslateLanguageDTO;
    }

    setRecordModelDTOList(recordModelDTOList) {
        this.recordModelDTOList = recordModelDTOList;
    }

    getRecordModelDTOList() {
        if (this.recordModelDTOList && !((this.recordModelDTOList instanceof Array && !(this.recordModelDTOList[0] instanceof BaseModel)))) {
            this.recordModelDTOList = ConvertUtil.ConvertGeneralWithArray(RecordModelDTO, this.recordModelDTOList);
        }
        return this.recordModelDTOList;
    }

    setRecordId(recordId) {
        this.recordId = recordId;
    }

    setCoreAllElementDTO(coreAllElementDTO) {
        this.coreAllElementDTO = coreAllElementDTO;
    }

    setCoreProcessRequestStatusDTO(coreProcessRequestStatusDTO) {
        this.coreProcessRequestStatusDTO = coreProcessRequestStatusDTO;
    }

    getCoreProcessRequestStatusDTO() {
        if (this.coreProcessRequestStatusDTO && !(this.coreProcessRequestStatusDTO instanceof BaseModel)) {
            this.coreProcessRequestStatusDTO = ConvertUtil.ConvertGeneral(CoreProcessRequestStatusDTO, this.coreProcessRequestStatusDTO);
        }
        return this.coreProcessRequestStatusDTO;
    }
}